jsonpb: panic unmarshaling private field

**Is your feature request related to a problem? Please describe.**

```
panic: reflect.Value.Interface: cannot return value obtained from unexported field or method [recovered]
	panic: reflect.Value.Interface: cannot return value obtained from unexported field or method [recovered]
	panic: reflect.Value.Interface: cannot return value obtained from unexported field or method

goroutine 6984 [running]:
panic({0x475c280, 0x5e70790})
	GOROOT/src/runtime/panic.go:941 +0x397 fp=0xc0098d12a8 sp=0xc0098d11e8 pc=0x48be77
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x3f6 fp=0xc0098d1448 sp=0xc0098d12a8 pc=0x2095556
runtime.deferCallSave(0xc0098d1518, 0xc004bb39f0?)
	GOROOT/src/runtime/panic.go:750 +0x82 fp=0xc0098d1458 sp=0xc0098d1448 pc=0x48ba82
runtime.runOpenDeferFrame(0xc0091f8050?, 0xc008461b30)
	GOROOT/src/runtime/panic.go:723 +0x1a5 fp=0xc0098d14a0 sp=0xc0098d1458 pc=0x48b8a5
panic({0x475c280, 0x5e70790})
	GOROOT/src/runtime/panic.go:838 +0x207 fp=0xc0098d1560 sp=0xc0098d14a0 pc=0x48bce7
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58 +0x3f6 fp=0xc0098d1700 sp=0xc0098d1560 pc=0x2095556
runtime.deferCallSave(0xc0098d17d0, 0xc004bb3918?)
	GOROOT/src/runtime/panic.go:750 +0x82 fp=0xc0098d1710 sp=0xc0098d1700 pc=0x48ba82
runtime.runOpenDeferFrame(0x5093b7?, 0xc0091f8050)
	GOROOT/src/runtime/panic.go:723 +0x1a5 fp=0xc0098d1758 sp=0xc0098d1710 pc=0x48b8a5
panic({0x475c280, 0x5e70790})
	GOROOT/src/runtime/panic.go:838 +0x207 fp=0xc0098d1818 sp=0xc0098d1758 pc=0x48bce7
reflect.valueInterface({0x475c280?, 0xc00910bf80?, 0x496a940?}, 0x10?)
	GOROOT/src/reflect/value.go:1441 +0xd8 fp=0xc0098d1858 sp=0xc0098d1818 pc=0x513b98
reflect.Value.Interface(...)
	GOROOT/src/reflect/value.go:1430
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc0098dcdc8, 0xc004f2ef00, {0x475c280?, 0xc00910bf80?, 0x32?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:568 +0x3f2 fp=0xc0098d1aa8 sp=0xc0098d1858 pc=0x9b9972
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f2ef00, {0x475c280?, 0xc00910bf80?, 0x475c280?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098d1b10 sp=0xc0098d1aa8 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5e97080?, 0xc00910bf80}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098d2160 sp=0xc0098d1b10 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f10600, {0x4e48640?, 0xc00910bf80?, 0xc004ba8408?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098d23b0 sp=0xc0098d2160 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f10600, {0x4e48640?, 0xc00910bf80?, 0x4e48640?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098d2418 sp=0xc0098d23b0 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5e9b010?, 0xc00910bf80}, {0x0, 0x0}, {0xc006f72dc0, 0x3d})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098d2a68 sp=0xc0098d2418 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalAny(0xc0074c3230, 0xc004bb2dc8, {0x5eaa610?, 0xc007650190?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:481 +0x77d fp=0xc0098d2b38 sp=0xc0098d2a68 pc=0x9b8d7d
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc004bb2dc8, {0x5eaa610?, 0xc007650190}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:217 +0x1fb fp=0xc0098d3188 sp=0xc0098d2b38 pc=0x9b60bb
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f10500, {0x4f66a00?, 0xc00910af58?, 0xc004ba9430?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098d33d8 sp=0xc0098d3188 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f10500, {0x4f66a00?, 0xc00910af58?, 0x4f66a00?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098d3440 sp=0xc0098d33d8 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5ea91d0?, 0xc00910af10}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098d3a90 sp=0xc0098d3440 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f10100, {0x4cedcc0?, 0xc00910af10?, 0xc004ba9d38?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098d3ce0 sp=0xc0098d3a90 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f10100, {0x4cedcc0?, 0xc00910af10?, 0x4cedcc0?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098d3d48 sp=0xc0098d3ce0 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5ea9200?, 0xc00910af00}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098d4398 sp=0xc0098d3d48 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f0fd00, {0x4e8c980?, 0xc000011300?, 0x43?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098d45e8 sp=0xc0098d4398 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f0fd00, {0x4e8c980?, 0xc000011300?, 0x4e8c980?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098d4650 sp=0xc0098d45e8 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5ea91a0?, 0xc00528d730}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098d4ca0 sp=0xc0098d4650 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f0fc00, {0x49fba20?, 0xc00528d730?, 0x35?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098d4ef0 sp=0xc0098d4ca0 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f0fc00, {0x49fba20?, 0xc00528d730?, 0x49fba20?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098d4f58 sp=0xc0098d4ef0 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5ea9290?, 0xc00528d730}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098d55a8 sp=0xc0098d4f58 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f0fb00, {0x4e8cac0?, 0xc000011308?, 0x49?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098d57f8 sp=0xc0098d55a8 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f0fb00, {0x4e8cac0?, 0xc000011308?, 0x4e8cac0?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098d5860 sp=0xc0098d57f8 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5ea91a0?, 0xc00528d6c0}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098d5eb0 sp=0xc0098d5860 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f0fa00, {0x49fba20?, 0xc00528d6c0?, 0x35?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098d6100 sp=0xc0098d5eb0 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f0fa00, {0x49fba20?, 0xc00528d6c0?, 0x49fba20?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098d6168 sp=0xc0098d6100 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5ea9290?, 0xc00528d6c0}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098d67b8 sp=0xc0098d6168 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f900, {0x4e8cac0?, 0xc000011310?, 0x49?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098d6a08 sp=0xc0098d67b8 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f900, {0x4e8cac0?, 0xc000011310?, 0x4e8cac0?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098d6a70 sp=0xc0098d6a08 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5ea91a0?, 0xc00528d650}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098d70c0 sp=0xc0098d6a70 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f800, {0x49fba20?, 0xc00528d650?, 0x35?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098d7310 sp=0xc0098d70c0 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f800, {0x49fba20?, 0xc00528d650?, 0x49fba20?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098d7378 sp=0xc0098d7310 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5ea9290?, 0xc00528d650}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098d79c8 sp=0xc0098d7378 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f700, {0x4e8cac0?, 0xc000011318?, 0x49?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098d7c18 sp=0xc0098d79c8 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f700, {0x4e8cac0?, 0xc000011318?, 0x4e8cac0?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098d7c80 sp=0xc0098d7c18 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5ea91a0?, 0xc00528d570}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098d82d0 sp=0xc0098d7c80 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f600, {0x49fba20?, 0xc00528d570?, 0x35?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098d8520 sp=0xc0098d82d0 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f600, {0x49fba20?, 0xc00528d570?, 0x49fba20?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098d8588 sp=0xc0098d8520 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5ea9290?, 0xc00528d570}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098d8bd8 sp=0xc0098d8588 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f500, {0x4e8cac0?, 0xc000011320?, 0x49?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098d8e28 sp=0xc0098d8bd8 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f500, {0x4e8cac0?, 0xc000011320?, 0x4e8cac0?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098d8e90 sp=0xc0098d8e28 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5ea91a0?, 0xc00528d490}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098d94e0 sp=0xc0098d8e90 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f400, {0x49fba20?, 0xc00528d490?, 0x35?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098d9730 sp=0xc0098d94e0 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f400, {0x49fba20?, 0xc00528d490?, 0x49fba20?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098d9798 sp=0xc0098d9730 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5ea9290?, 0xc00528d490}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098d9de8 sp=0xc0098d9798 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f300, {0x4e8cac0?, 0xc000011328?, 0x49?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098da038 sp=0xc0098d9de8 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f300, {0x4e8cac0?, 0xc000011328?, 0x4e8cac0?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098da0a0 sp=0xc0098da038 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5ea91a0?, 0xc00528d420}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098da6f0 sp=0xc0098da0a0 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f200, {0x49fba20?, 0xc00528d420?, 0x35?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098da940 sp=0xc0098da6f0 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f200, {0x49fba20?, 0xc00528d420?, 0x49fba20?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098da9a8 sp=0xc0098da940 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5ea9290?, 0xc00528d420}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098daff8 sp=0xc0098da9a8 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f100, {0x4e8cac0?, 0xc000011330?, 0x49?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098db248 sp=0xc0098daff8 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f100, {0x4e8cac0?, 0xc000011330?, 0x4e8cac0?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098db2b0 sp=0xc0098db248 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5ea91a0?, 0xc0061759d0}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098db900 sp=0xc0098db2b0 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f000, {0x4f2f5c0?, 0xc005231128?, 0xc004bb1ba8?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098dbb50 sp=0xc0098db900 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f0f000, {0x4f2f5c0?, 0xc005231128?, 0x4f2f5c0?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098dbbb8 sp=0xc0098dbb50 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5e94c20?, 0xc005231100}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098dc208 sp=0xc0098dbbb8 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc004bb2dc8, 0xc004f0eb00, {0x4e5fe80?, 0xc0000112f8?, 0x203002?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:657 +0xb25 fp=0xc0098dc458 sp=0xc0098dc208 pc=0x9ba0a5
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalValue(0xc0074c3230, 0xc0098dcdc8, 0xc004f0eb00, {0x46daf80?, 0xc006bdbab0?, 0xc004bb2700?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:543 +0x21f9 fp=0xc0098dc6a8 sp=0xc0098dc458 pc=0x9bb779
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalField(0xc0074c3230, 0xc004bb2dc8, 0xc004f0eb00, {0x46daf80?, 0xc006bdbab0?, 0x46daf80?}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:513 +0x3e7 fp=0xc0098dc710 sp=0xc0098dc6a8 pc=0x9b9507
github.com/gogo/protobuf/jsonpb.(*Marshaler).marshalObject(0xc0074c3230, 0xc0098dcdc8, {0x5e93db0?, 0xc006bdb9e0}, {0x0, 0x0}, {0x0, 0x0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:371 +0x1c65 fp=0xc0098dcd60 sp=0xc0098dc710 pc=0x9b7b25
github.com/gogo/protobuf/jsonpb.(*Marshaler).Marshal(0x0?, {0x5e71080, 0xc0074c3260}, {0x5e93db0?, 0xc006bdb9e0?})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:138 +0x1b5 fp=0xc0098dcdf8 sp=0xc0098dcd60 pc=0x9b5c55
github.com/gogo/protobuf/jsonpb.(*Marshaler).MarshalToString(0x0?, {0x5e93db0, 0xc006bdb9e0})
	github.com/gogo/protobuf/jsonpb/external/com_github_gogo_protobuf/jsonpb/jsonpb.go:144 +0x56 fp=0xc0098dce40 sp=0xc0098dcdf8 pc=0x9b5d36
github.com/cockroachdb/cockroach/pkg/sql/protoreflect.marshalToJSON({0x5ee69e8?, 0xc006bdb9e0}, 0x0)
	github.com/cockroachdb/cockroach/pkg/sql/protoreflect/utils.go:106 +0xeb fp=0xc0098dcf00 sp=0xc0098dce40 pc=0x1551c6b
github.com/cockroachdb/cockroach/pkg/sql/protoreflect.MessageToJSON({0x5ee69e8?, 0xc006bdb9e0?}, {0x80?, 0xec?})
	github.com/cockroachdb/cockroach/pkg/sql/protoreflect/utils.go:118 +0x39 fp=0xc0098dcf28 sp=0xc0098dcf00 pc=0x1551db9
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.glob..func198.1({0xc005eac690?, 0x9700d77?}, {0xc00482ec80?, 0x0?, 0x0?}, {0x45?, 0x0?})
	github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/builtins.go:3953 +0x49 fp=0xc0098dcf80 sp=0xc0098dcf28 pc=0x20e2fc9
github.com/cockroachdb/cockroach/pkg/sql/sem/builtins.glob..func198.3(0x5f09fe0?, {0xc0071a48d0, 0x3, 0xc005be5728?})
	github.com/cockroachdb/cockroach/pkg/sql/sem/builtins/builtins.go:3987 +0x132 fp=0xc0098dcff8 sp=0xc0098dcf80 pc=0x20e38d2
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*defaultBuiltinFuncOperator).Next.func1()
	github.com/cockroachdb/cockroach/pkg/sql/colexec/builtin_funcs.go:77 +0x13c fp=0xc0098dd0a8 sp=0xc0098dcff8 pc=0x28cd71c
github.com/cockroachdb/cockroach/pkg/sql/colmem.(*Allocator).PerformOperation(0xc005242ff0?, {0xc0098dd150, 0x1, 0x0?}, 0xc004bb3160?)
	github.com/cockroachdb/cockroach/pkg/sql/colmem/allocator.go:349 +0x49 fp=0xc0098dd0f8 sp=0xc0098dd0a8 pc=0x2780da9
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*defaultBuiltinFuncOperator).Next(0xc0076f2640)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/builtin_funcs.go:54 +0x185 fp=0xc0098dd1c0 sp=0xc0098dd0f8 pc=0x28cd5a5
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase.(*simpleProjectOp).Next(0xc008f1e1e0)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase/simple_project.go:124 +0x3f fp=0xc0098dd280 sp=0xc0098dd1c0 pc=0x286eb7f
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils.(*vectorTypeEnforcer).Next(0xc005243040)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils/operator.go:138 +0x2e fp=0xc0098dd2c8 sp=0xc0098dd280 pc=0x279476e
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecprojconst.projJSONFetchValJSONBytesConstOp.Next({{{{{...}}, {{...}}}, 0xc0071a4a80, 0x0, 0x3, 0x0}, {0xc001e208bf, 0x9, 0x9}})
	github.com/cockroachdb/cockroach/bazel-out/k8-opt/bin/pkg/sql/colexec/colexecprojconst/proj_const_right_ops.eg.go:22331 +0x3c fp=0xc0098dd390 sp=0xc0098dd2c8 pc=0x2e4817c
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecprojconst.(*projJSONFetchValJSONBytesConstOp).Next(0x202?)
	<autogenerated>:1 +0x98 fp=0xc0098dd450 sp=0xc0098dd390 pc=0x2f3ad18
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils.(*vectorTypeEnforcer).Next(0xc005243090)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils/operator.go:138 +0x2e fp=0xc0098dd498 sp=0xc0098dd450 pc=0x279476e
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecprojconst.projJSONFetchValJSONBytesConstOp.Next({{{{{...}}, {{...}}}, 0xc0071a4a50, 0x3, 0x4, 0x0}, {0xc001e208c8, 0xe, 0xe}})
	github.com/cockroachdb/cockroach/bazel-out/k8-opt/bin/pkg/sql/colexec/colexecprojconst/proj_const_right_ops.eg.go:22331 +0x3c fp=0xc0098dd560 sp=0xc0098dd498 pc=0x2e4817c
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecprojconst.(*projJSONFetchValJSONBytesConstOp).Next(0x5072397?)
	<autogenerated>:1 +0x98 fp=0xc0098dd620 sp=0xc0098dd560 pc=0x2f3ad18
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecsel.(*selEQJSONJSONConstOp).Next(0xc008c0e4c0)
	github.com/cockroachdb/cockroach/bazel-out/k8-opt/bin/pkg/sql/colexec/colexecsel/selection_ops.eg.go:9061 +0x48 fp=0xc0098dd728 sp=0xc0098dd620 pc=0x2f82b48
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase.(*simpleProjectOp).Next(0xc008f1e420)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase/simple_project.go:124 +0x3f fp=0xc0098dd7e8 sp=0xc0098dd728 pc=0x286eb7f
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase.(*simpleProjectOp).Next(0xc008f1e480)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase/simple_project.go:124 +0x3f fp=0xc0098dd8a8 sp=0xc0098dd7e8 pc=0x286eb7f
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).next(0xc00742dc20)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/materializer.go:303 +0x73 fp=0xc0098dd8d0 sp=0xc0098dd8a8 pc=0x28d35d3
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).nextAdapter(0xc00742dc20)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/materializer.go:328 +0x1e fp=0xc0098dd8e8 sp=0xc0098dd8d0 pc=0x28d377e
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).nextAdapter-fm()
	<autogenerated>:1 +0x26 fp=0xc0098dd900 sp=0xc0098dd8e8 pc=0x295ca06
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0x0?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x62 fp=0xc0098dd940 sp=0xc0098dd900 pc=0x2095102
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Materializer).Next(0xc00742dc20)
	github.com/cockroachdb/cockroach/pkg/sql/colexec/materializer.go:334 +0x4c fp=0xc0098dd978 sp=0xc0098dd940 pc=0x28d382c
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).next(0xc005f52000)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:130 +0x3b fp=0xc0098dd9a8 sp=0xc0098dd978 pc=0x31431fb
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).nextAdapter(0xc005f52000)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:147 +0x1e fp=0xc0098dd9c0 sp=0xc0098dd9a8 pc=0x31432be
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).nextAdapter-fm()
	<autogenerated>:1 +0x26 fp=0xc0098dd9d8 sp=0xc0098dd9c0 pc=0x3151386
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0x0?)
	github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:92 +0x62 fp=0xc0098dda18 sp=0xc0098dd9d8 pc=0x2095102
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*FlowCoordinator).Next(0xc005f52000)
	github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:152 +0x3e fp=0xc0098dda60 sp=0xc0098dda18 pc=0x314337e
github.com/cockroachdb/cockroach/pkg/sql/execinfra.Run({0x5eb1af0, 0xc008c0e340}, {0x5ed21d0, 0xc005f52000}, {0x5e8a2c0?, 0xc0068b5880?})
	github.com/cockroachdb/cockroach/pkg/sql/execinfra/base.go:186 +0x55 fp=0xc0098ddab0 sp=0xc0098dda60 pc=0x2249bf5
created by github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).runWithEx
	github.com/cockroachdb/cockroach/pkg/sql/internal.go:213 +0x195
```

This is in the context of this upgrade query:
https://github.com/cockroachdb/cockroach/blob/de12cee396210cf905a0677b5ecc76773a14990a/pkg/upgrade/upgrades/wait_for_del_range_in_gc_job.go#L43-L44


**Additional context**
Root cause of https://github.com/cockroachdb/cockroach/issues/86509

Jira issue: CRDB-18863

--- a/pkg/kv/kvclient/kvcoord/dist_sender_test.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender_test.go
@@ -3791,12 +3791,11 @@ func TestErrorIndexAlignment(t *testing.T) {
 			var testFn simpleSendFn = func(ctx context.Context, ba roachpb.BatchRequest) (*roachpb.BatchResponse, error) {
 				reply := ba.CreateReply()
 				if nthRequest == tc.nthPartialBatch {
-					reply.Error = &roachpb.Error{
-						// The relative index is always 0 since
-						// we return an error for the first
-						// request of the nthPartialBatch.
-						Index: &roachpb.ErrPosition{Index: 0},
-					}
+					reply.Error = roachpb.NewErrorf("foo")
+					// The relative index is always 0 since
+					// we return an error for the first
+					// request of the nthPartialBatch.
+					reply.Error.Index = &roachpb.ErrPosition{Index: 0}
 				}
 				nthRequest++
 				return reply, nil

--- a/pkg/kv/kvserver/client_merge_test.go
+++ b/pkg/kv/kvserver/client_merge_test.go
@@ -2223,7 +2223,7 @@ func TestStoreRangeMergeCheckConsistencyAfterSubsumption(t *testing.T) {
 	close(abortMergeTxn)
 
 	pErr := <-mergeErr
-	require.IsType(t, &roachpb.Error{}, pErr)
+	require.IsType(t, (*roachpb.Error)(nil), pErr)
 	require.Regexp(t, "abort the merge for test", pErr.String())
 
 	testutils.SucceedsSoon(t, func() error {

--- a/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/deadlocks
+++ b/pkg/kv/kvserver/concurrency/testdata/concurrency_manager/deadlocks
@@ -168,7 +168,7 @@ on-txn-updated txn=txn1 status=aborted
 [-] update txn: aborting txn1
 [4] sequence req1r: detected pusher aborted
 [4] sequence req1r: conflicted with 00000002-0000-0000-0000-000000000000 on "b" for 0.000s
-[4] sequence req1r: sequencing complete, returned error: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): <nil>
+[4] sequence req1r: sequencing complete, returned error: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED)
 [6] sequence req3r: resolving intent "a" for txn 00000001 with ABORTED status
 [6] sequence req3r: lock wait-queue event: done waiting
 [6] sequence req3r: conflicted with 00000001-0000-0000-0000-000000000000 on "a" for 0.000s
@@ -402,7 +402,7 @@ on-txn-updated txn=txn1 status=aborted
 [4] sequence req4w: sequencing complete, returned guard
 [5] sequence req1w2: detected pusher aborted
 [5] sequence req1w2: conflicted with 00000002-0000-0000-0000-000000000000 on "b" for 0.000s
-[5] sequence req1w2: sequencing complete, returned error: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): <nil>
+[5] sequence req1w2: sequencing complete, returned error: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED)
 [7] sequence req3w2: resolving intent "a" for txn 00000001 with ABORTED status
 [7] sequence req3w2: lock wait-queue event: wait for (distinguished) txn 00000004 running request @ key "a" (queuedWriters: 1, queuedReaders: 0)
 [7] sequence req3w2: conflicted with 00000001-0000-0000-0000-000000000000 on "a" for 0.000s
@@ -651,7 +651,7 @@ on-txn-updated txn=txn4 status=aborted
 [-] update txn: aborting txn4
 [4] sequence req4w: detected pusher aborted
 [4] sequence req4w: conflicted with 00000003-0000-0000-0000-000000000000 on "c" for 0.000s
-[4] sequence req4w: sequencing complete, returned error: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): <nil>
+[4] sequence req4w: sequencing complete, returned error: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED)
 [5] sequence req1w2: lock wait-queue event: done waiting
 [5] sequence req1w2: conflicted with 00000004-0000-0000-0000-000000000000 on "b" for 0.000s
 [5] sequence req1w2: acquiring latches
@@ -889,7 +889,7 @@ on-txn-updated txn=txn1 status=aborted
 [-] update txn: aborting txn1
 [5] sequence req1w2: detected pusher aborted
 [5] sequence req1w2: conflicted with 00000004-0000-0000-0000-000000000000 on "b" for 0.000s
-[5] sequence req1w2: sequencing complete, returned error: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): <nil>
+[5] sequence req1w2: sequencing complete, returned error: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED)
 [6] sequence req3w2: resolving intent "a" for txn 00000001 with ABORTED status
 [6] sequence req3w2: lock wait-queue event: done waiting
 [6] sequence req3w2: conflicted with 00000001-0000-0000-0000-000000000000 on "a" for 0.000s
@@ -1154,7 +1154,7 @@ on-txn-updated txn=txn4 status=aborted
 [-] update txn: aborting txn4
 [5] sequence req4w: detected pusher aborted
 [5] sequence req4w: conflicted with 00000005-0000-0000-0000-000000000000 on "b" for 0.000s
-[5] sequence req4w: sequencing complete, returned error: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED): <nil>
+[5] sequence req4w: sequencing complete, returned error: TransactionAbortedError(ABORT_REASON_PUSHER_ABORTED)
 [6] sequence req3w2: lock wait-queue event: done waiting
 [6] sequence req3w2: conflicted with 00000004-0000-0000-0000-000000000000 on "a" for 0.000s
 [6] sequence req3w2: acquiring latches

--- a/pkg/kv/kvserver/txnwait/queue_test.go
+++ b/pkg/kv/kvserver/txnwait/queue_test.go
@@ -222,6 +222,8 @@ func TestMaybeWaitForPushWithContextCancellation(t *testing.T) {
 	cancel()
 	pErr := <-waitingRes
 	require.NotNil(t, pErr)
+	s := pErr.String()
+	_ = s
 	require.Regexp(t, context.Canceled.Error(), pErr)
 	require.Equal(t, 0, q.mu.txns[txn.ID].waitingPushes.Len())
 

--- a/pkg/roachpb/BUILD.bazel
+++ b/pkg/roachpb/BUILD.bazel
@@ -52,7 +52,6 @@ go_library(
         "//pkg/util/uuid",
         "@com_github_cockroachdb_apd_v3//:apd",
         "@com_github_cockroachdb_errors//:errors",
-        "@com_github_cockroachdb_errors//errorspb",
         "@com_github_cockroachdb_errors//extgrpc",
         "@com_github_cockroachdb_redact//:redact",
         "@com_github_dustin_go_humanize//:go-humanize",

--- a/pkg/roachpb/errors.go
+++ b/pkg/roachpb/errors.go
@@ -22,7 +22,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
 	"github.com/cockroachdb/cockroach/pkg/util/uuid"
 	"github.com/cockroachdb/errors"
-	"github.com/cockroachdb/errors/errorspb"
 	_ "github.com/cockroachdb/errors/extgrpc" // register EncodeError support for gRPC Status
 	"github.com/cockroachdb/redact"
 )
@@ -153,26 +152,6 @@ func NewError(err error) *Error {
 	e := &Error{
 		EncodedError: errors.EncodeError(context.Background(), err),
 	}
-
-	// This block is deprecated behavior retained for compat with
-	// 20.2 nodes. It makes sure that if applicable, deprecatedMessage,
-	// ErrorDetail, and deprecatedTransactionRestart are set.
-	{
-		if intErr, ok := err.(*internalError); ok {
-			*e = *(*Error)(intErr)
-		} else if detail := ErrorDetailInterface(nil); errors.As(err, &detail) {
-			e.deprecatedMessage = detail.message(e)
-			if r, ok := detail.(transactionRestartError); ok {
-				e.deprecatedTransactionRestart = r.canRestartTransaction()
-			} else {
-				e.deprecatedTransactionRestart = TransactionRestart_NONE
-			}
-			e.deprecatedDetail.MustSetInner(detail)
-			e.checkTxnStatusValid()
-		} else {
-			e.deprecatedMessage = err.Error()
-		}
-	}
 	return e
 }
 
@@ -202,42 +181,7 @@ func (e *Error) SafeFormat(s redact.SafePrinter, _ rune) {
 		return
 	}
 
-	if e.EncodedError != (errors.EncodedError{}) {
-		err := errors.DecodeError(context.Background(), e.EncodedError)
-		var iface ErrorDetailInterface
-		if errors.As(err, &iface) {
-			// Deprecated code: if there is a detail and the message produced by the detail
-			// (which gets to see the surrounding Error) is different, then use that message.
-			// What is likely the cause of that is that someone passed an updated Transaction
-			// to the Error via SetTxn, and the error detail prints that txn.
-			//
-			// TODO(tbg): change SetTxn so that instead of stashing the transaction on the
-			// Error struct, it wraps the EncodedError with an error containing the updated
-			// txn. Make GetTxn retrieve the first one it sees (overridden by UnexposedTxn
-			// while it's still around). Remove the `message(Error)` methods from ErrorDetailInterface.
-			// We also have to remove GetDetail() in the process since it doesn't understand the
-			// wrapping; instead we need a method that looks for a specific kind of detail, i.e.
-			// we basically want to use `errors.As` instead.
-			deprecatedMsg := iface.message(e)
-			if deprecatedMsg != err.Error() {
-				s.Print(deprecatedMsg)
-				return
-			}
-		}
-		s.Print(err)
-	} else {
-		// TODO(tbg): remove this block in the 21.2 cycle and rely on EncodedError
-		// always being populated.
-		switch t := e.GetDetail().(type) {
-		case nil:
-			s.Print(e.deprecatedMessage)
-		default:
-			// We have a detail and ignore e.deprecatedMessage. We do assume that if a detail is
-			// present, e.deprecatedMessage does correspond to that detail's message. This
-			// assumption is not enforced but appears sane.
-			s.Print(t)
-		}
-	}
+	s.Print(errors.DecodeError(context.Background(), e.EncodedError))
 
 	if txn := e.GetTxn(); txn != nil {
 		s.SafeString(": ")
@@ -252,15 +196,11 @@ func (e *Error) String() string {
 
 // TransactionRestart returns the TransactionRestart for this Error.
 func (e *Error) TransactionRestart() TransactionRestart {
-	if e.EncodedError == (errorspb.EncodedError{}) {
-		// Legacy code.
-		//
-		// TODO(tbg): delete in 21.2.
-		return e.deprecatedTransactionRestart
-	}
-	var iface transactionRestartError
-	if errors.As(errors.DecodeError(context.Background(), e.EncodedError), &iface) {
-		return iface.canRestartTransaction()
+	if e.EncodedError.IsSet() {
+		var iface transactionRestartError
+		if errors.As(errors.DecodeError(context.Background(), e.EncodedError), &iface) {
+			return iface.canRestartTransaction()
+		}
 	}
 	return TransactionRestart_NONE
 }
@@ -345,7 +285,7 @@ func (e *Error) GoError() error {
 	if e == nil {
 		return nil
 	}
-	if e.EncodedError != (errorspb.EncodedError{}) {
+	if e.EncodedError.IsSet() {
 		err := errors.DecodeError(context.Background(), e.EncodedError)
 		var iface transactionRestartError
 		if errors.As(err, &iface) {
@@ -374,18 +314,11 @@ func (e *Error) GoError() error {
 
 // GetDetail returns an error detail associated with the error, or nil otherwise.
 func (e *Error) GetDetail() ErrorDetailInterface {
-	if e == nil {
+	if e == nil || !e.EncodedError.IsSet() {
 		return nil
 	}
 	var detail ErrorDetailInterface
-	if e.EncodedError != (errorspb.EncodedError{}) {
-		errors.As(errors.DecodeError(context.Background(), e.EncodedError), &detail)
-	} else {
-		// Legacy behavior.
-		//
-		// TODO(tbg): delete in v21.2.
-		detail, _ = e.deprecatedDetail.GetInner().(ErrorDetailInterface)
-	}
+	errors.As(errors.DecodeError(context.Background(), e.EncodedError), &detail)
 	return detail
 }
 
@@ -407,30 +340,21 @@ func (e *Error) UpdateTxn(o *Transaction) {
 	} else {
 		e.UnexposedTxn.Update(o)
 	}
-	if sErr, ok := e.deprecatedDetail.GetInner().(ErrorDetailInterface); ok {
-		// Refresh the message as the txn is updated.
-		//
-		// TODO(tbg): deprecated, remove in 21.2.
-		e.deprecatedMessage = sErr.message(e)
-	}
 	e.checkTxnStatusValid()
 }
 
 // checkTxnStatusValid verifies that the transaction status is in-sync with the
 // error detail.
 func (e *Error) checkTxnStatusValid() {
-	// TODO(tbg): this will need to be updated when we
-	// remove all of these deprecated fields in 21.2.
-
 	txn := e.UnexposedTxn
-	err := e.deprecatedDetail.GetInner()
+	err := e.GetDetail()
 	if txn == nil {
 		return
 	}
-	if e.deprecatedTransactionRestart == TransactionRestart_NONE {
+	if errors.HasType(err, (*TransactionAbortedError)(nil)) {
 		return
 	}
-	if errors.HasType(err, (*TransactionAbortedError)(nil)) {
+	if e.TransactionRestart() == TransactionRestart_NONE {
 		return
 	}
 	if txn.Status.IsFinalized() {

--- a/pkg/roachpb/errors.proto
+++ b/pkg/roachpb/errors.proto
@@ -674,17 +674,6 @@ message ErrPosition {
 message Error {
   option (gogoproto.goproto_stringer) = false;
 
-  // message is a human-readable error message.
-  //
-  // DEPRECATED.
-  optional string message = 1 [(gogoproto.nullable) = false, (gogoproto.customname) = "deprecatedMessage"];
-
-  // If transaction_restart is not NONE, the error condition may be handled by
-  // restarting the transaction.
-  //
-  // DEPRECATED.
-  optional TransactionRestart transaction_restart = 3 [(gogoproto.nullable) = false, (gogoproto.customname) = "deprecatedTransactionRestart"];
-
   // An optional updated transaction. This is to be used by the client in case
   // of retryable errors.
   //
@@ -693,12 +682,6 @@ message Error {
 
   // Node at which the error was generated (zero if does not apply).
   optional int32 origin_node = 5 [(gogoproto.nullable) = false, (gogoproto.casttype) = "NodeID"];
-
-  // If an ErrorDetail is present, it may contain additional structured data
-  // about the error.
-  //
-  // DEPRECATED - consult encoded_error instead.
-  optional ErrorDetail detail = 6 [(gogoproto.nullable) = false, (gogoproto.customname) = "deprecatedDetail"];
 
   // encoded_error is the Go error that caused this Error.
   optional errorspb.EncodedError encoded_error = 9 [(gogoproto.nullable) = false];
@@ -712,7 +695,7 @@ message Error {
   optional util.hlc.Timestamp now = 8 [(gogoproto.nullable) = false,
     (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/util/hlc.ClockTimestamp"];
 
-  reserved 2;
+  reserved 1, 2, 3, 6 ;
 }
 
 

--- a/pkg/roachpb/errors_test.go
+++ b/pkg/roachpb/errors_test.go
@@ -89,7 +89,7 @@ func TestErrorTxn(t *testing.T) {
 	if txn := pErr.GetTxn(); txn != nil {
 		t.Fatalf("wanted nil, unexpected: %+v", txn)
 	}
-	pErr = &Error{}
+	pErr = NewErrorf("foo")
 	const name = "X"
 	pErr.SetTxn(&Transaction{Name: name})
 	if txn := pErr.GetTxn(); txn == nil || txn.Name != name {
@@ -155,49 +155,6 @@ func TestErrorRedaction(t *testing.T) {
 		act := s.RedactableString().Redact()
 		const exp = "ReadWithinUncertaintyIntervalError: read at time 0.000000001,0 encountered previous write with future timestamp 0.000000002,0 within uncertainty interval `t <= (local=0.000000002,2, global=0.000000003,0)`; observed timestamps: [{12 0.000000004,0}]: \"foo\" meta={id=00000000 key=‹×› pri=0.00005746 epo=0 ts=0.000000001,0 min=0.000000001,0 seq=0} lock=true stat=PENDING rts=0.000000001,0 wto=false gul=0.000000002,0"
 		require.Equal(t, exp, string(act))
-	})
-}
-
-func TestErrorDeprecatedFields(t *testing.T) {
-	// Verify that deprecated fields are populated and queried correctly.
-
-	t.Run("unstructured", func(t *testing.T) {
-		err := errors.New("I am an error")
-		pErr := NewError(err)
-		pErr.EncodedError.Reset()
-
-		require.Equal(t, err.Error(), pErr.String())
-		require.IsType(t, &internalError{}, pErr.GoError())
-		require.Equal(t, err.Error(), pErr.GoError().Error())
-		require.Equal(t, err.Error(), pErr.deprecatedMessage)
-		require.Equal(t, TransactionRestart_NONE, pErr.deprecatedTransactionRestart)
-		require.Nil(t, pErr.deprecatedDetail.Value)
-	})
-	txn := MakeTransaction("foo", Key("k"), 0, hlc.Timestamp{WallTime: 1}, 50000, 99)
-
-	t.Run("structured-wrapped", func(t *testing.T) {
-		// For extra spice, wrap the structured error. This ensures
-		// that we populate the deprecated fields even when
-		// the error detail is not the head of the error chain.
-		err := NewReadWithinUncertaintyIntervalError(
-			hlc.Timestamp{WallTime: 1},
-			hlc.Timestamp{WallTime: 2},
-			hlc.Timestamp{WallTime: 2, Logical: 2},
-			&txn,
-		)
-
-		pErr := NewError(errors.Wrap(err, "foo"))
-		// Quick check that the detail round-trips when EncodedError is still there.
-		require.EqualValues(t, err, pErr.GetDetail())
-
-		pErr.EncodedError.Reset()
-
-		var ure *UnhandledRetryableError
-		require.True(t, errors.As(pErr.GoError(), &ure))
-		require.Equal(t, &ure.PErr, pErr)
-		require.Contains(t, pErr.GoError().Error(), err.Error())
-		require.EqualValues(t, err, pErr.GetDetail())
-		require.Equal(t, TransactionRestart_IMMEDIATE, pErr.deprecatedTransactionRestart)
 	})
 }
 

